esmvalcore/preprocessor/_rolling_window.py

```diff
@@ -0,0 +1,51 @@
+"""Rolling-window operations on data cubes."""
+
+import logging
+
+import iris
+import iris.exceptions
+
+from ._shared import get_iris_analysis_operation
+
+logger = logging.getLogger(__name__)
+
+
+def rolling_window_statistics(cube, coordinate, operator, window_length):
+    """Compute rolling-window statistics over a coordinate.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube
+        Input cube.
+    coordinate : str
+        Coordinate over which the rolling-window statistics is calculated.
+    operator : str
+        Select operator to apply. Available operators: ``'mean'``,
+        ``'median'``, ``'std_dev'``, ``'sum'``, ``'variance'``, ``'min'``,
+        ``'max'``.
+    window_length : int
+        Size of the window to use.
+
+    Returns
+    -------
+    iris.cube.Cube
+        Rolling-window statistics cube.
+
+    Raises
+    ------
+    iris.exceptions.CoordinateNotFoundError:
+        Cube does not have time coordinate.
+    ValueError:
+        Invalid ``'operator'`` given.
+    """
+    try:
+        cube.coord(coordinate)
+    except iris.exceptions.CoordinateNotFoundError:
+        logger.error("Cube %s does not have %s coordinate", cube, coordinate)
+        raise
+
```

Comment:
```suggestion
```

This can be removed, ESMValTool automatically issues an error message with all necessary information for each preprocessor that fails, e.g.:

```bash
2022-08-24 07:14:53,007 UTC [827825] ERROR   Failed to run preprocessor function 'rolling_window_statistics' on the data                                                                                                                                                       
[<iris 'Cube' of air_temperature / (K) (time: 324; latitude: 64; longitude: 128)>]
loaded from original input file(s)
['/work/bd0854/DATA/ESMValTool2/CMIP5_DKRZ/CCCma/CanESM2/historical/mon/atmos/Amon/r1i1p1/v20120718/tas/tas_Amon_CanESM2_historical_r1i1p1_185001-200512.nc']
with function argument(s)
coordinate = 'timei',
operator = 'mean',
window_length = 3
```